File: occasum-core/src/main/java/pw/stamina/occasum/node/instructions/FolderPropertyNodeInstruction.java
package pw.stamina.occasum.node.instructions;

import pw.stamina.occasum.PropertyHandle;
import pw.stamina.occasum.node.PropertyNode;

final class FolderPropertyNodeInstruction implements PropertyNodeInstruction {
    private final PropertyHandle handle;
    private final String name;

    FolderPropertyNodeInstruction(PropertyHandle handle, String name) {
        this.handle = handle;
        this.name = name;
    }

    @Override
    public void apply(PropertyNode node) {
        if (handle != null) {
            node.folder(handle, name);
        } else {
            node.folder(name);
        }
    }
}

File: occasum-core/src/main/kotlin/pw/stamina/occasum/node/instructions/FolderPropertyNodeInstruction.kt
package pw.stamina.occasum.node.instructions

import pw.stamina.occasum.PropertyHandle
import pw.stamina.occasum.node.PropertyNode

internal class FolderPropertyNodeInstruction(
        private val handle: PropertyHandle?,
        private val name: String)
    : PropertyNodeInstruction {

    override fun apply(node: PropertyNode) {
        if (handle != null) {
            node.folder(handle, name)
        } else {
            node.folder(name)
        }
    }
}
